test/sampler_test.rb

```diff
@@ -95,4 +95,35 @@ def test_tracer_with_rate_sampler
       assert_equal(0.5, span.get_metric(Datadog::RateSampler::SAMPLE_RATE_METRIC_KEY))
     end
   end
+
+  def test_priority_sampler_handling_existing_priority
+    tracer = get_test_tracer
+    tracer.provider.context = Datadog::Context.new
+    tracer.configure(sampler: Datadog::PrioritySampler.new)
+
+    tracer.provider.context.sampling_priority = Datadog::Ext::Priority::USER_KEEP
+    tracer.trace('test_keep', trace_id: 1) {}
+
+    tracer.provider.context.sampling_priority = Datadog::Ext::Priority::AUTO_KEEP
+    tracer.trace('test_keep', trace_id: 2) {}
+
+    tracer.provider.context.sampling_priority = Datadog::Ext::Priority::AUTO_REJECT
+    tracer.trace('test_reject', trace_id: 3) {}
+
+    tracer.provider.context.sampling_priority = Datadog::Ext::Priority::USER_REJECT
+    tracer.trace('test_reject', trace_id: 4) {}
+
+    assert_equal(2, tracer.writer.spans.length)
```

Comment:
For brevity, this is okay right now, but when this gets refactored into RSpec, we should assert each one did as expected. Otherwise this test could pass if it were doing the complete opposite that we want, because there would still be two spans.

Reply:
Yeah. I've just added a test te verify all spans are named 'test_keep'.

Also this would have probably been nicer to write in RSpec. I chose to keep minitest only to shorten the dev-time.